pkg/instance/network.go

```diff
@@ -119,34 +119,22 @@ func (n *network) AddPortUDP(port int) error {
 }
 
 // GetIP returns the IP of the instance
-// This function can only be called in the states 'Preparing' and 'Started'
+// This function can only be called in the states 'Started'
 func (n *network) GetIP(ctx context.Context) (string, error) {
```

Comment:
The problem may be that the developer expects this IP to be static. But in fact, the IP changes if the Pod restarts. A restart can happen either by the users' intentions or 'randomly' by Kubernetes.
One idea is to document it properly, but the developer might oversee this. Another option would be to remove `GetIP` and entirely require the switch to `HostName.`

Reply:
We should definitely document the fact that the IP can change.

Reply:
The function documentation makes sense. But I know the devs will not read that and then get confused that it does not work as expected.
Currently, I don't see any case where they need the IP. Therefore, I vote to remove the `GetIP` function.

Reply:
There is also GetServiceIP. 

Removing one or both of those methods would create a lot of bloat to this PR. 

If we want to remove that I recommend we do that in a separate PR.

Reply:
`GetServiceIP` can't return an IP if it's `headless`.

Reply:
Maybe we can change the `GetIP` name to something else like `GetEphemeralIP` and for `GetServiceIP`, i think there is a specific error when it is headless, if not I will add it.

Reply:
I checked it, there is a particular error when user requests GetServiceIP on a headless service